python/lsst/obs/base/gen2to3/translators.py

```diff
@@ -276,55 +278,27 @@ def extract(self, gen2id, *args, **kwargs):
         return self._map.get(abstract, abstract)
 
 
-class Translator:
-    """Callable object that translates Gen2 Data IDs to Gen3 Data IDs for a
-    particular DatasetType.
-
-    Translators should usually be constructed via the `makeMatching` method.
-
-    Parameters
-    ----------
-    handlers : `list`
-        A list of KeyHandlers this Translator should use.
-    skyMap : `BaseSkyMap`, optional
-        SkyMap instance used to define any tract or patch Dimensions.
-    skyMapName : `str`
-        Gen3 SkyMap Dimension name to be associated with any tract or patch
-        Dimensions.
-    datasetTypeName : `str`
-        Name of the dataset type whose data IDs this translator handles.
-    """
-    def __init__(self, handlers: List[KeyHandler], skyMap: Optional[BaseSkyMap], skyMapName: Optional[str],
-                 datasetTypeName: str):
-        self.handlers = handlers
-        self.skyMap = skyMap
-        self.skyMapName = skyMapName
-        self.datasetTypeName = datasetTypeName
-
-    __slots__ = ("handlers", "skyMap", "skyMapName", "datasetTypeName")
+class TranslatorFactory:
 
-    # Rules used to match Handlers when constring a Translator.
-    # outer key is instrument name, or None for any
-    # inner key is DatasetType name, or None for any
-    # values are 3-tuples of (frozenset(gen2keys), handler, consume)
-    _rules: Dict[
-        Optional[str],
-        Dict[
+    def __init__(self):
+        # Rules used to match Handlers when constring a Translator.
+        # outer key is instrument name, or None for any
+        # inner key is DatasetType name, or None for any
+        # values are 3-tuples of (frozenset(gen2keys), handler, consume)
```

Comment:
I think this comment should live in the class docstring, so it's actually in the docs.

Reply:
I would absolutely agree if this was a public attribute, but I don't think private attributes belong in public docs.

Reply:
In that case, it's not obvious what specifically the comments are describing. Can that be made more clear?

Reply:
I realized I could actually put these next to the type annotation components they refer to in the lines below, which makes it even more clear.